Incorrect type checking of ScalarFloorDivNode operands

If I correctly understand the following line :
https://github.com/flexflow/FlexFlow/blob/ad627c9dd7cac82f7894320a8d896b3b55e6b2dc/python/flexflow/torch/model.py#L1706
scalar floor division supports only int or float operands. 

Then the check is erroneous. We should crash when `scalar` is not an int *and* is not a float.

@sydneyhauke, the current check is indeed erroneous and will always return True. Thanks for pointing it out.

@lockshaw, I'll send in a PR. Meanwhile, some observations/questions on code style:

1. `assert 0` can usually be replaced by some more meaningful error check or raising more informative exceptions. Google style guide: https://google.github.io/styleguide/pyguide.html#244-decision and an example:
```
$ python error.py
  File "/home/soumya/error.py", line 4, in <module>
    f()
  File "/home/soumya/error.py", line 2, in f
    assert 0, "Error message"
AssertionError: Error message

$ python error.py
Traceback (most recent call last):
  File "/home/soumya/error.py", line 4, in <module>
    f()
  File "/home/soumya/error.py", line 2, in f
    assert 5==3, "Error message"
AssertionError: Error message
```
2. Do we have a perferred way to checking types? `isinstance(x, y)` vs `type(x) == y`

--- a/python/flexflow/torch/model.py
+++ b/python/flexflow/torch/model.py
@@ -1703,7 +1703,7 @@ class ScalarFloorDivNode(FunctionNode):
     def parse(self):
         s = [self.name]
         scalar = self.innodes[1]
-        if type(scalar) is not int or type(scalar) is not float:
+        if type(scalar) not in [int, float]:
             assert 0, "FlexFlow does not support tensor floor division"
         innodes = (self.innodes[0],)
         s.append(self.parse_inoutnodes(innodes))

--- a/python/flexflow/torch/model.py
+++ b/python/flexflow/torch/model.py
@@ -1703,7 +1703,7 @@ class ScalarFloorDivNode(FunctionNode):
     def parse(self):
         s = [self.name]
         scalar = self.innodes[1]
-        if type(scalar) not in [int, float]:
+        if not isinstance(scalar, [int, float]):
             assert 0, "FlexFlow does not support tensor floor division"
         innodes = (self.innodes[0],)
         s.append(self.parse_inoutnodes(innodes))

